p2panda-rs/src/secret_group/codec.rs

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::convert::TryFrom;
+use std::io::{Read, Write};
+
+use openmls::framing::MlsCiphertext;
+use tls_codec::{TlsDeserialize, TlsSerialize, TlsSize};
+
+use crate::secret_group::lts::LongTermSecretCiphertext;
+use crate::secret_group::SecretGroupMessage;
+
+/// `SecretGroupMessageType` is an additional "helper" enum next to the `SecretGroupMessage` enum.
+/// It is used as the first byte to distict the type of the inner message data during TLS en- /
+/// decoding.
+#[derive(Debug, Clone, Copy, TlsSerialize, TlsDeserialize, TlsSize)]
+#[repr(u8)]
+enum SecretGroupMessageType {
+    /// This message contains user data encrypted and encoded in form of a MLS application message.
+    MlsApplicationMessage = 1,
```

Comment:
Good idea! I would not use `Schema` though, as this might be misleading in the p2panda world. How about:

SenderRatchetMode
LongTermSecretMode

.. or setting? or encryption?

Reply:
> I would not use Schema though

I was not thinking straight, I did not intend to write that :D Too many schemas. Sorry!

Let's try again. Right now, a secret group message can be one of two things, based on how it was encrypted:

- either `SecretGroupMessage::MLSApplicationMessage`
- or `SecretGroupMessage::LongTermSecretMessage`

I think it would be clearer if the naming reflected the differentiation, i.e. the encryption. So that would be:

- `SecretGroupMessage::SenderRatchetSecretMessage`
- `SecretGroupMessage::LongTermSecretMessage`

In fact, as "message" is already contained in `SecretGroupMessage` it would be even better to go with:

- `SecretGroupMessage::SenderRatchetSecret`
- `SecretGroupMessage::LongTermSecret`